klog.go

```diff
@@ -802,10 +803,16 @@ func (l *loggingT) infoS(logger *logr.Logger, filter LogFilter, depth int, msg s
 // set log severity by s
 func (l *loggingT) printS(err error, s severity, depth int, msg string, keysAndValues ...interface{}) {
 	b := &bytes.Buffer{}
-	b.WriteString(fmt.Sprintf("%q", msg))
+	// The message is never quoted. New lines get indented.
+	writeString(b, msg)
+	// Enhance readability by inserting : between message and key/value
```

Comment:
> I don't agree with this change,

This PR or the line you commented on, the `:` separator? I can drop that separator, it's not important.

But quoting is a major problem. I get what was meant to be achieved with it, but IMHO the effect on readability was not considered. it now prevents conversion to structured logging because developers will complain or even go as far as [filing bugs](https://github.com/kubernetes/kubernetes/issues/104868) and [reverting the conversion](https://github.com/kubernetes/kubernetes/pull/106228).

Reply:
I agree with Marek, JSON should not become a dependency for log parsing, the raw log output should be parsable

Reply:
The current text format is not machine-readable. If that was the goal, then we have bugs in its implementation around handling of structs.

Reply:
So as you are both against this proposal, how do you suggest we should deal with cases where developers want multi-line output in a readable format (https://github.com/kubernetes/kubernetes/issues/104868)?

Reply:
I think we should separate how we consider multi line for different part of logs. In my opinion multiline:
* **doesn't make sense for log messages**. They should be a short description of what event is happening in system. Their format should not differ much or at all between different log formats. For example as I user that switched between log formats I want to be able to find log I have seen in old format in the new format just by grepping the message. 
* **doesn't make sense for log keys**. They should strive to uniquely identify types of metadata attached to log. Strive comes from idea that over time a schema will develop that will have a predictable way to find values under specific keys. Their names should follow a strict convention to avoid diverging.
* **can make sense for complex values when debugging**. I assumed handling this case is needed kubernetes/kubernetes#104868. There are multiple cases in k8s logs that multiline logs are used to improve readability. We should allow users to customize that some values (for example dumping whole k8s object yaml) are written in multiline format.

Reply:
> can make sense for complex values when debugging. 

Agreed.

> We should allow users to customize that some values (for example dumping whole k8s object yaml) are written in multiline format.

And how to you propose to do that?

We need and API for it *and* a serialization format that log parsers then will understand.

Reply:
> In my opinion multiline doesn't make sense for log messages.

I can buy that argument. So let me propose a solution just for multi-line values in https://github.com/kubernetes/klog/pull/273

I'm going to close this PR here in favor of that one.

/close